From 748f2dbf8ff222465dfdd1db4884bafbcd8c6ffd Mon Sep 17 00:00:00 2001
From: zhang sanshan <b51434@freescale.com>
Date: Mon, 27 Jul 2015 15:55:41 +0800
Subject: [PATCH] MA-6924 [Android_MX6_MX7] fastboot: 'fastboot
 reboot-bootloader' do not work

'fastboot reboot-bootloader' is took as 'fastboot reboot' in V2015.04.
The length of fastboot command is decided by the length of cmd_dispatch_info's cmd.
So need put reboot-bootloader in front of reboot.

Signed-off-by: zhang sanshan <b51434@freescale.com>
---
 drivers/usb/gadget/f_fastboot.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/drivers/usb/gadget/f_fastboot.c b/drivers/usb/gadget/f_fastboot.c
index 002704f..4f3d0f8 100644
--- a/drivers/usb/gadget/f_fastboot.c
+++ b/drivers/usb/gadget/f_fastboot.c
@@ -2344,6 +2344,12 @@ struct cmd_dispatch_info {
 };
 
 static const struct cmd_dispatch_info cmd_dispatch_info[] = {
+#ifdef CONFIG_FSL_FASTBOOT
+	{
+		.cmd = "reboot-bootloader",
+		.cb = cb_reboot_bootloader,
+	},
+#endif
 	{
 		.cmd = "reboot",
 		.cb = cb_reboot,
@@ -2373,12 +2379,6 @@ static const struct cmd_dispatch_info cmd_dispatch_info[] = {
 		.cmd = "oem",
 		.cb = cb_oem,
 	},
-#ifdef CONFIG_FSL_FASTBOOT
-	{
-		.cmd = "reboot-bootloader",
-		.cb = cb_reboot_bootloader,
-	},
-#endif
 };
 
 static void rx_handler_command(struct usb_ep *ep, struct usb_request *req)
-- 
1.8.0

